packages/block-editor/src/components/block-list/style.scss

```diff
@@ -56,15 +56,6 @@
 		margin-left: -$block-padding;
 		margin-right: -$block-padding;
 	}
-
-	// Space every block, and the default appender, using margins.
-	// This allows margins to collapse, which gives a better representation of how it looks on the frontend.
-	.block-editor-default-block-appender > .block-editor-default-block-appender__content,
```

Comment:
I don't know if having the margins here is a bad thing. I do understand the reasoning behind moving them to the individual block to allow them to be overridden there. But were there really a lot of complaints about the margins not matching up with the front-end?
Like I've said before, I think a little variance is expected in the administrative UI. Enough space for controls to work is absolutely a good reason for this.

That said, I think the space could be reduced from `32px`. 
The element's margins will take over as long as they're greater than the `.block-editor-block-list__block-edit` margins.

For example if a paragraph's `p` tag has a `margin-bottom: 40px`, then that block will have `40px` margin between it and the next block in the editor. If it has `margin-bottom: 16px`, then it'll have the "fallback" margin from `.block-editor-block-list__block-edit` of `32px`. 
So the trick is to define the minimum required margin for a good editing experience and let theme styles go bigger but not smaller.

In my short experimenting `24px` seems to work well. (wish it was less, but it just isn't)
`24px` roughly equates to `1.5rem` which is a common margin, I think, so that's good. I would avoid `em` because of it's unpredictability.

And @ellatrix has a good point. The top margin is a bit pointless. Only place it would be seen is on the `:first-child`, but a `margin-bottom` on `.editor-post-title` would accomplish the same.

Reply:
> But were there really a lot of complaints about the margins not matching up with the front-end?

Well, see feedback on this thread ;) 

> Like I've said before, I think a little variance is expected in the administrative UI. Enough space for controls to work is absolutely a good reason for this.

The ultimate goal is for the editor to be able to look exactly like the frontend with minimal work by the theme editor. Every research we've conducted suggests that when there's a strong visual connection between the two, usability increases. I agree, a little variance is acceptable, but I also feel the more we can do to reduce this variance, the better. The difficult balance is to ensure a good editing experience for themes that do not style the editor.

To the rest of your feedback, very solid, thank you, and I agree with Ella too. Will take a stab at addressing both of your points.

Reply:
I tried to chagne the 32px to 24px. But it starts to affect the multi-select appearance. 

32px:

![Screenshot 2019-03-29 at 11 11 52](https://user-images.githubusercontent.com/1204802/55226483-60014e80-5215-11e9-9417-b5f6eb9974a3.png)

24px:

![Screenshot 2019-03-29 at 11 15 56](https://user-images.githubusercontent.com/1204802/55226494-6394d580-5215-11e9-8cb9-8cf4fd1d6c22.png)

What happens there is that every block has a footprint, and _outside_ that footprint, the block borders are drawn. By default this block border is drawn 14px out, which means two blocks next to each other, both multi-selected, will have 14px + 14px of "selection rectangle" drawn around them. Which means if the margin between the two blocks was 28px, they would appear flush next to each other. At 32px there's  4px gap. At 24px, there's a 4px overlap which is what you're seeing in the second screenshot.

All that is to say: _we can change that_. There's a $block-padding variable that's trivial to change. If 24px is the magic number, we can set the block padding to 10 or 12px. 

But this is a big visual change, and I would love to ship this PR before exploring such a change. Do you think we could explore such a margin and padding change separately?

Reply:
You've been busy today @jasmussen ! :smile: 

> it was about the selector being too specific and therefore hard for themes to override.

You're right. In my idealistic mind, themes shouldn't need to think about overriding anything and selectors like `.block-editor-block-list__block-edit` shouldn't ever be a part of a theme's vocabulary. 
We're lucky enough to have a lot of semantic HTML5 element wrappers on our blocks. These elements are often the first things to get styled in themes, old and new. 
```
p,
ul,
blockquote,
table,
figure {
    margin: 0 0 1rem;
}
```
That's free backward compatibility if we can take advantage of it.

> Do you think we could explore such a margin and padding change separately?

Definitely. I think setting the minimum spacing on this wrapping editor element will be key to making this work well. It will act as a safety net in case somebody is using a crazy Meyer type reset but it'll allow a themes vertical rhythm to shine through in most cases.

Reply:
> Definitely. I think setting the minimum spacing on this wrapping editor element will be key to making this work well. It will act as a safety net in case somebody is using a crazy Meyer type reset but it'll allow a themes vertical rhythm to shine through in most cases.

I think this is probably a key insight. @kjellr suggested something slightly similar in a recent chat. In https://github.com/WordPress/gutenberg/pull/14407#issuecomment-477953187 I tried a generic rule that didn't quite with out. But maybe this idea can inspire a better rule?